File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        int x = str1.size();
        int y = str2.size();
        int period = gcd(x, y);
        string res = "";
        for(int i = 0; i < period; i++) res += str1[i];
        bool f = true;
        for(int i = 0; i < x; i++){
            if(str1[i] != res[i%period]){
             f = false; 
            } 
        }
        if(!f){
            res = "";
        }
        f = true;
        for(int i = 0; i < y; i++){
            if(str2[i] != res[i%period]){
             f = false; 
            } 
        }
        if(!f){
            res = "";
        }
        return res;
    }
};

File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.cs
public class Solution {
    public string GcdOfStrings(string str1, string str2) {
        int x = str1.Length;
        int y = str2.Length;
        int period = GCD(x, y);
        string res = "";
        for(int i = 0; i < period; i++) res += str1[i];
        if(str1 + str2 != str2 + str1) res = "";
        return res;
    }
    public int GCD(int a, int b){
        return Math.Abs(b == 0 ? a : GCD(b, a % b));
    }
}